services/worker-manager/src/data.js

```diff
@@ -246,7 +248,7 @@ class Worker {
   // UNIQUE_VIOLATION when those checks fail.
   async create(db) {
```

Comment:
It looks like a worker is never created with a non-null secret, so perhaps there's no need for a `create_worker_2` function either?